Turn off colors for any output

We should avoid coloring Terraform's output as this causes color escape codes to pollute the error output:

```
--- FAIL: TestAccPacketOperatingSystem_NotFound (2.45s)
    testing_new.go:35: WARNING: destroy failed, so remote objects may still exist and be subject to billing
    testing_new.go:35: failed to destroy: terraform failed: exit status 1
        
        stderr:
         [31m
         [1m [31mError:  [0m [0m [1mThere are no operating systems that match the search criteria [0m
        
         [0m  on config452693439/terraform_plugin_test.tf line 2, in data "packet_operating_system" "example":
           2: 	data "packet_operating_system" "example"  [4m{ [0m
         [0m
         [0m [0m
        
FAIL
```

Maybe longer-term we'd want to explore machine-readable diagnostics, but that would require a lot more effort investment in Terraform itself.

--- a/helper.go
+++ b/helper.go
@@ -135,8 +135,9 @@ func (h *Helper) NewWorkingDir() (*WorkingDir, error) {
 	}
 
 	return &WorkingDir{
-		h:       h,
-		baseDir: dir,
+		h:        h,
+		baseArgs: []string{"-no-color"},
+		baseDir:  dir,
 	}, nil
 }
 

--- a/working_dir.go
+++ b/working_dir.go
@@ -19,6 +19,9 @@ type WorkingDir struct {
 	// baseDir is the root of the working directory tree
 	baseDir string
 
+	// baseArgs is arguments that should be appended to all commands
+	baseArgs []string
+
 	// configDir contains the singular config file generated for each test
 	configDir string
 }
@@ -111,7 +114,10 @@ func (wd *WorkingDir) RequireClearPlan(t TestControl) {
 }
 
 func (wd *WorkingDir) init(pluginDir string) error {
-	return wd.runTerraform("init", "-plugin-dir="+pluginDir, wd.configDir)
+	args := []string{"init"}
+	args = append(args, wd.baseArgs...)
+	args = append(args, "-plugin-dir="+pluginDir, wd.configDir)
+	return wd.runTerraform(args...)
 }
 
 // Init runs "terraform init" for the given working directory, forcing Terraform
@@ -163,7 +169,10 @@ func (wd *WorkingDir) planFilename() string {
 // CreatePlan runs "terraform plan" to create a saved plan file, which if successful
 // will then be used for the next call to Apply.
 func (wd *WorkingDir) CreatePlan() error {
-	return wd.runTerraform("plan", "-out=tfplan", wd.configDir)
+	args := []string{"plan"}
+	args = append(args, wd.baseArgs...)
+	args = append(args, "-out=tfplan", wd.configDir)
+	return wd.runTerraform(args...)
 }
 
 // RequireCreatePlan is a variant of CreatePlan that will fail the test via
@@ -182,12 +191,15 @@ func (wd *WorkingDir) RequireCreatePlan(t TestControl) {
 // plan and apply it.
 func (wd *WorkingDir) Apply() error {
 	args := []string{"apply"}
+	args = append(args, wd.baseArgs...)
+
 	if wd.HasSavedPlan() {
 		args = append(args, "tfplan")
 	} else {
 		args = append(args, "-auto-approve")
 		args = append(args, wd.configDir)
 	}
+
 	return wd.runTerraform(args...)
 }
 
@@ -207,7 +219,10 @@ func (wd *WorkingDir) RequireApply(t TestControl) {
 // If destroy fails then remote objects might still exist, and continue to
 // exist after a particular test is concluded.
 func (wd *WorkingDir) Destroy() error {
-	args := []string{"destroy", "-auto-approve", wd.configDir}
+	args := []string{"destroy"}
+	args = append(args, wd.baseArgs...)
+
+	args = append(args, "-auto-approve", wd.configDir)
 	return wd.runTerraform(args...)
 }
 
@@ -242,7 +257,10 @@ func (wd *WorkingDir) SavedPlan() (*tfjson.Plan, error) {
 	}
 
 	var ret tfjson.Plan
-	err := wd.runTerraformJSON(&ret, "show", "-json", wd.planFilename())
+	args := []string{"show"}
+	args = append(args, wd.baseArgs...)
+	args = append(args, "-json", wd.planFilename())
+	err := wd.runTerraformJSON(&ret, args...)
 	if err != nil {
 		return nil, err
 	}
@@ -269,7 +287,10 @@ func (wd *WorkingDir) RequireSavedPlan(t TestControl) *tfjson.Plan {
 // If the state cannot be read, State returns an error.
 func (wd *WorkingDir) State() (*tfjson.State, error) {
 	var ret tfjson.State
-	err := wd.runTerraformJSON(&ret, "show", "-json")
+	args := []string{"show"}
+	args = append(args, wd.baseArgs...)
+	args = append(args, "-json")
+	err := wd.runTerraformJSON(&ret, args...)
 	if err != nil {
 		return nil, err
 	}
@@ -293,7 +314,9 @@ func (wd *WorkingDir) RequireState(t TestControl) *tfjson.State {
 
 // Import runs terraform import
 func (wd *WorkingDir) Import(resource, id string) error {
-	args := []string{"import", "-config=" + wd.configDir, resource, id}
+	args := []string{"import"}
+	args = append(args, wd.baseArgs...)
+	args = append(args, "-config="+wd.configDir, resource, id)
 	return wd.runTerraform(args...)
 }
 
@@ -310,6 +333,7 @@ func (wd *WorkingDir) RequireImport(t TestControl, resource, id string) {
 // Refresh runs terraform refresh
 func (wd *WorkingDir) Refresh() error {
 	args := []string{"refresh"}
+	args = append(args, wd.baseArgs...)
 	return wd.runTerraform(args...)
 }
 

